scalems/wrappers/__init__.py

```diff
@@ -16,32 +15,109 @@
 
 Core components:
     make_input(**kwargs) -> SimulationInput:
+
         A simulation tool module provides a *make_input* command that accepts
         implementation-dependent named arguments and returns an object that may
         be used as simulation input within the context of the other module tools.
 
+        Proposal: At an abstract level, this contains everything needed to define the simulation 
+        mathematically. The chemical system, instructions for running the simulation, but 
+        not how it's executed, or the resources it will take. That will be taken care of later in
+        the workflow.
+
+        GROMACS: .mpd,.top,.gro
+        OpenMM: (look at ExTasy workflows)
+        PLUMED-GROMACS: GROMACS + PLUMED input file
+        LAMMPS: .input file
+
+        Crucially, it provide a place for the information to be staged
+        for modifying input, such that a single simulation can be
+        loaded, and then variants of that simulation can be "cloned out", and modified.
+
+        For many programs, this may simply be putting the text of the
+        files into a container where they can be retrived and
```

Comment:
Per previous discussions, I think we agreed it was reasonable to have an abstract text object with fields that can be filled by a task, such as with the Python `str.format()`. See #37 

It is a little more work (or maybe just awkward) to impose the `make_input()->modify_input()` calls in such a case, and I would submit that the further back we go from `simulate() -> Simulation`, the less firmly we can specify things right now, and `make_input()` is really a placeholder for future work.

We can do it, though, by requiring that the SimulationInput contains both the Text future and sources of substitution to use if not replaced by a `modify_input`.